### PR TITLE
Support for raw and multi-line strings.

### PR DESCRIPTION
**Pull request by Josef-Vonasek**
### Pull Request Description
Fixes rewind bug. Adds support for raw and multi-line strings.

### Important Notes
Added `EOL(validIndent: Boolean)`, which signifies end of line. In case of multi line block `validIndent = false` when the next line is wrongly indented. 

### Checklist

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by Josef-Vonasek** [comment]
Can I link github issue to this PR somehow?

**Comment by iamrecursion** [comment]
You'll need the ZenHub browser extension installed (or go to the ZenHub site). It should then provide you an option to "connect this pull request with an existing issue" which you can use to do exactly that.

File: syntax/specialization/src/test/scala/org/enso/syntax/text/Parser.scala [hunk 1]
```diff
@@ -64,126 +60,144 @@ class ParserSpec extends FlatSpec with Matchers {
...
+//  //// UTF Surrogates ////
+//  ////////////////////////
+//
+//  "\uD800\uDF1E" ?= Unrecognized("\uD800\uDF1E")
+//
+//  /////////////////////
+//  //// Large Input ////
+//  /////////////////////
+//
+//  "BIG_INPUT_" * ParserBase.BUFFERSIZE ?= "BIG_INPUT_" * ParserBase.BUFFERSIZE
```
Comment (wdanilo) [inline syntax/specialization/src/test/scala/org/enso/syntax/text/Parser.scala]: Why the tests are disabled?

File: syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala [hunk 2]
```diff
@@ -358,28 +364,75 @@ object AST {
...
+          case s         => List(s)
+        }
+
+      def _dup(quote: Quote, segments: List[Segment]) =
+        copy(indent, quoteChar, quote, segments)
+    }
+
+    object MultiLine {
+
+      def preprocess(indent: Int, rawSegments: List[Segment]): List[Segment] = {
```
Comment (wdanilo) [inline syntax/definition/src/main/scala/org/enso/syntax/text/AST.scala]: Lets rename `preprocess` to anything with a semantic meaning. What does exactly this function do?

File: lib/flexer/src/main/scala/org/enso/flexer/ParserBase.scala [hunk 3]
```diff
@@ -141,14 +141,21 @@ trait ParserBase[T] {
...
+    rewind(charsToLastRule + 1)
+    matchBuilder.setLength(matchBuilder.length - charsToLastRule)
+    call(rule)
+  }
+
+  final def call(rule: () => Unit): Int = {
+    currentMatch = matchBuilder.result()
     charsToLastRule = 0
+    rule()
+    -1
```
[the remark this answers is not in the document]
Comment (Josef-Vonasek) [reply lib/flexer/src/main/scala/org/enso/flexer/ParserBase.scala]: The next state after running any group_rule is -1

File: syntax/specialization/src/test/scala/org/enso/syntax/text/Parser.scala (hunk 1 above)
Comment (Josef-Vonasek) [reply syntax/specialization/src/test/scala/org/enso/syntax/text/Parser.scala]: Opps.

**Pull request merged**